engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/component/ComponentExtractor.scala

```diff
@@ -0,0 +1,71 @@
+package pl.touk.nussknacker.engine.component
+
+import com.typesafe.config.Config
+import com.vdurmont.semver4j.Semver
+import net.ceedubs.ficus.Ficus._
+import pl.touk.nussknacker.engine.api.component._
+import pl.touk.nussknacker.engine.api.process._
+import pl.touk.nussknacker.engine.component.ComponentExtractor.componentConfigPath
+import pl.touk.nussknacker.engine.util.loader.ScalaServiceLoader
+import pl.touk.nussknacker.engine.version.BuildInfo
+
+object ComponentExtractor {
+
+  val componentConfigPath = "components"
+
+  def apply(classLoader: ClassLoader): ComponentExtractor = {
+    val version = NussknackerVersion(new Semver(BuildInfo.version))
+    ComponentExtractor(classLoader, version)
+  }
+
+}
+
+case class ComponentExtractor(classLoader: ClassLoader, nussknackerVersion: NussknackerVersion) {
+
+  private val providers = ScalaServiceLoader.load[ComponentProvider](classLoader).map(p => p.providerName -> p).toMap
+
+  private def loadCorrectComponents(config: Config): Map[String, (ComponentProviderConfig, ComponentProvider)] = {
+    import net.ceedubs.ficus.readers.ArbitraryTypeReader._
+    val componentsConfig = config.getAs[Map[String, ComponentProviderConfig]](componentConfigPath).getOrElse(Map.empty)
+    componentsConfig.filterNot(_._2.disabled).map {
+      case (name, providerConfig: ComponentProviderConfig) =>
+        val providerName = providerConfig.providerType.getOrElse(name)
+        val provider = providers.getOrElse(providerName, throw new IllegalArgumentException(s"Provider $providerName not found"))
+        if (!provider.isCompatible(nussknackerVersion)) {
+          throw new IllegalArgumentException(s"Component provider $name (of type $providerName) is not compatible with $nussknackerVersion, please disable it explicitly ")
```

Comment:
"please use correct component provider version or disable it explicitly"

Reply:
:+1: